### PR TITLE
DF-1347 Better Input Validation

### PR DESCRIPTION
**Pull request by sebworks**
DF-1347 Better Input Validation

## Changes
 
- Updated  ' _includes/post-macros.html' to add validation class to markup.
- Updated  '  static/js/app.js' to validate pagination form on submission.

## Notes
- This does not handle cases where the user has modified the query parameters directly
- @anselmbradford, I'll convert to CommonJS if #499 goes in first.

## Test
- Navigate to http://localhost:7000/blog/ and enter 'a'. 
- Click 'Go' (Should do nothing).
- Enter '5' (Should navigate to 'http://localhost:7000/blog/?page=5#pagination_content') .

## Review
 
- @jimmynotjim  
- @anselmbradford 
- @Scotchester  
- @KimberlyMunoz

### CODE REVIEW

File: _includes/post-macros.html [hunk 1]
```diff
@@ -250,6 +250,8 @@ <h1 class="h4">{{ icon }} {{ header }}</h1>
                    id="pagination_current-page"
                    name="page"
                    type="number"
+                   inputmode="numeric"
+                   pattern="[0-9]*"
```
Comment (jimmynotjim) [inline _includes/post-macros.html]: Nice addition!

File: static/js/app.js [hunk 2]
```diff
@@ -487,6 +487,33 @@ $('.js-validate-filters').each(function() {
 });
 
 /* ==========================================================================
+   Pagination Form validation
+
+   Check to verify page input value is number and with min/max range
+   before submitting form.
+   ========================================================================== */
+
+$('.js-validate_pagination').each(function() {
```
Comment (jimmynotjim) [inline static/js/app.js]: Let's add white space to wrapping parens to match what we've been doing elsewhere.

**Comment by anselmbradford** [comment]
LGTM pull in https://github.com/cfpb/cfgov-refresh/pull/499

**Comment by anselmbradford** [comment]
Also see https://github.com/cfpb/cfgov-refresh/pull/497#issuecomment-98778358 for reference on migrating JS.

**Pull request closed without merging**